### PR TITLE
Added methods to convert Ed25519 keys to Curve25519 keys.

### PR DESCRIPTION
**Pull request by pavel-mukhanov**
Bindings for libsodium methods:
```
crypto_sign_ed25519_pk_to_curve25519(...)
crypto_sign_ed25519_sk_to_curve25519(...)
```

### CODE REVIEW

**Review by popzxc — APPROVED** [review]

File: src/crypto/sign/ed25519.rs [hunk 1]
```diff
@@ -149,6 +152,35 @@ pub fn verify_detached(
     }
 }
 
+/// Converts Ed25519 keypair to Curve25519 keyipair.
+pub fn convert_ed_keys_to_curve25519(
+    pk: PublicKey,
+    sk: SecretKey
+) -> ([u8; SCALARMULTBYTES], [u8; SCALARMULTBYTES]) {
+    let mut pk = clone_into_array(&pk[..]);
```
Comment (aleksuss) [inline src/crypto/sign/ed25519.rs]: Should `pk` and `sk` be as mutable ? As I can see `crypto_sign_ed25519_sk_to_curve25519` accepts constant pointer to array ?

Comment (pavel-mukhanov) [reply src/crypto/sign/ed25519.rs]: You are right, they shouldn't.

**Review by stanislav-tkach — APPROVED** [review]

File: src/crypto/sign/ed25519.rs [hunk 2]
```diff
@@ -149,6 +152,35 @@ pub fn verify_detached(
     }
 }
 
+/// Converts Ed25519 keypair to Curve25519 keyipair.
+pub fn convert_ed_keys_to_curve25519(
```
Comment (vldm) [inline src/crypto/sign/ed25519.rs]: You probably need a seperate function for public and secret key too

**Review by vldm — APPROVED** [review]

**Review by aleksuss — APPROVED** [review]
LGTM

**Review by aleksuss — APPROVED** [review]

**Pull request merged**